packages/pds/src/db/index.ts

```diff
@@ -2,62 +2,18 @@ import assert from 'assert'
 import { Kysely, SqliteDialect, PostgresDialect, Migrator } from 'kysely'
 import SqliteDB from 'better-sqlite3'
 import { Pool as PgPool, types as pgTypes } from 'pg'
-import { ValidationError } from '@atproto/lexicon'
-import * as Declaration from './records/declaration'
-import * as Post from './records/post'
-import * as Vote from './records/vote'
-import * as Repost from './records/repost'
-import * as Trend from './records/trend'
-import * as Follow from './records/follow'
-import * as Assertion from './records/assertion'
-import * as Confirmation from './records/confirmation'
-import * as Profile from './records/profile'
-import { AtUri } from '@atproto/uri'
-import * as common from '@atproto/common'
-import { CID } from 'multiformats/cid'
-import { dbLogger as log } from '../logger'
 import { DatabaseSchema } from './database-schema'
-import * as scrypt from './scrypt'
-import { User } from './tables/user'
 import { dummyDialect } from './util'
 import * as migrations from './migrations'
 import { CtxMigrationProvider } from './migrations/provider'
-import { DidHandle } from './tables/did-handle'
-import { Record as DeclarationRecord } from '../lexicon/types/app/bsky/system/declaration'
-import { APP_BSKY_GRAPH } from '../lexicon'
-import { MessageQueue } from './types'
 
 export class Database {
```

Comment:
With this class so stripped down now & most application logic moved to the services, would it make sense to have this `extend Kysely<DatabaseSchema>`?

This would keep us from having to do the `db.db` nonsense that we do everywhere.

Really this class serves to:
- give static methods for creation
- simplify the tx flow a bit & manage some state about if we're in a tx or not
- give migration helpers

I think we could just ditch the `close` fn in favor of Kysely's `destroy`

Reply:
Love this idea, and sadly striking out on it right now.  I'd like to came back to it, though.  The issue is that I'm not coming up with a straightforward way to get an instance of our extended class for transactions.  I.e. hard to upgrade the `txn` that comes out of kysely here: `this.transaction().execute((txn) => {`, which would mess with some things like checking the db dialect in a transaction, asserting that the instance is a transaction, etc.

I think it's figureoutable, just might be a bit more involved than I can pull off with the other stuff going on in this particular PR.

Reply:
cool cool no worries, it's not pressing 👌